internal/http/services/owncloud/ocdav/trashbin.go

```diff
@@ -469,6 +469,26 @@ func (h *TrashbinHandler) restore(w http.ResponseWriter, r *http.Request, s *svc
 		return
 	}
 
+	// Restoring to a non-existent location is not supported by the WebDAV spec. The following block ensures the target
+	// restore location exists, and if it doesn't returns a conflict error code.
+	if dstStatRes.Status.Code == rpc.Code_CODE_NOT_FOUND && strings.HasSuffix(dst, itemPath) && isNested(dst) {
```

Comment:
`strings.HasSuffix(dst, itemPath)` do we always need to check this? Shouldn't we check that the parent dir exists even if the name of the file is to be changed?

Reply:
That sounds correct; I think this was a leftover. Let me update it and run CI as soon as possible 🥇